### ISSUE TITLE
[Bug] PEP 563 + `typing.NamedTuple` = `BeartypeCallHintForwardRefException` :woozy_face: 

### ISSUE THREAD

**Issue by kasium** [beartype/beartype#318]
**Module some.py**
```py
from __future__ import annotations
import typing

class Tup(typing.NamedTuple):
    val: int

TUP = Tup(123)
```
**Execute:** 
`python -c "from beartype.claw import beartype_package;  beartype_package('some'); import some"`

**Error:**
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "some.py", line 10, in <module>
    Tup(123)
  File "<@beartype(namedtuple_Tup.Tup.__new__) at 0x7f927d72f740>", line 21, in __new__
  File "venv/lib/python3.12/site-packages/beartype/_check/forward/fwdtype.py", line 326, in __missing__
    hint = import_module_attr(
           ^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/beartype/_util/module/utilmodimport.py", line 199, in import_module_attr
    raise exception_cls(
beartype.roar.BeartypeCallHintForwardRefException: Forward reference "namedtuple_Tup.int" unimportable.
```

The error does not happen if `from __future__ import annotations` is removed.
I use beartype 0.17.0 (great release btw) and python 3.12.0

**Comment by leycec** [comment]
**Yikes!** Thanks so much for the kind words and horrifying heads-up, @kasium. I now regret playing video games and enjoying my birthday all weekend. *@beartype, how could PEP 563 still be so broken?* :woozy_face: 

That's such an odd exception, too. Clearly, there's no `namedtuple_Tup` class in sight. What *is* that thing? Something horrifying. That's what. I'll dig deep into this on Monday and push out a new release resolving this "shortly" thereafter. Until then, please tell @beartype to stop being dumb by blacklisting `NamedTuple` subclasses: <sup>i'm so sorry</sup>

```python
from __future__ import annotations
import typing

# Make the bad @beartyping go away.
from beartype import BeartypeConf, BeartypeStrategy
quietbear = beartype(conf=BeartypeConf(strategy=BeartypeStrategy.O0))

@quietbear
class Tup(typing.NamedTuple):
    val: int

TUP = Tup(123)
```

Completely untested. Pretend that does something.

**Comment by kasium** [comment]
@leycec thanks for the fast actions! Please don't regret taking some time off, @beartype is an awesome project and I'm happy to report issues 😄 Let's see how you solve it

**Comment by leycec** [comment]
**Resolved by db7d538befea.** Aaaand... that was shockingly *brutal.* That was a whole new level of brutality best exemplified by nested genre tags on RateYourMusic like "obscure progressive technical war death metal from Quebec." Let's pretend that never happened.

Now, if you'll excuse me, I have a years-long backlog of turn-based role playing games to complete. You're up next, "Final Fantasy IX: Moguri Mod." You'd better live up to the hype, too! :worried:

**Comment by kasium** [comment]
Thanks for the fast fix. Have fun 👍
